04week/mastermind.js

```diff
@@ -28,21 +28,52 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-  // your code here
+
+//Display "0-1", "1-3" as appropriate. Compare the guess to the solution and generate what should be populated
+//Use sting.split('') to convert the string into an array of comparison if needed
+//Assign a null value where the first value in a guess string matches a value in the solution string
+function generateHint(guess) {
+  let exactMatches = 0;
+  let fuzzyMatches = 0;
+  let guessArr = guess.split('');
+  let solnArr = solution.split('');
+
+  for (let i=0; i < 4; i++) {
+    //exact matches
+    if (guessArr[i] == solnArr[i]) {
+      exactMatches = exactMatches + 1;
+      guessArr[i] = null;
+      solnArr[i] = null;
+    //fuzzy matches
+    } else if (solnArr.includes(guess[i]) && guessArr.includes(solution[i]) && guessArr[i] != solnArr[i]) {
```

Comment:
Doing it this way, a ball might be counted twice, ocne for the fuzzy match, and once for a subsequent exact match. 
This would happen if the solution had a repeating color